buddyboss/buddyboss-platform
# PROD-728 Link Preview of Publicly added url is redirected to Login Screen

### All Submissions:






### Changes proposed in this Pull Request:



Fixes #3029

**Describe the bug** 
when site is made private with privacy setting checked and some url is added as public, user is taken to login screen if link is accessed anonymously 
![image](https://user-images.githubusercontent.com/80830022/127309129-35866bf7-a5d3-404b-b956-a55b09150e9f.png)

API
```
curl -H "accept:application/json, text/plain, */*" -H "keep-alive:true" -H "connection:keep-alive" -H "accesstoken:eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczpcL1wvYmItYXBwLXBsYXRmb3JtLmZseXdoZWVsc2l0ZXMuY29tOnJlc3QtYXBpIiwiaWF0IjoxNjI3NDU1NjAxLCJuYmYiOjE2Mjc0NTU2MDEsImV4cCI6MTYzNTIzMTYwMSwianRpIjoibEUpXkxHdUpUcUs5UHFFZCM4RnZAIzRXeGQ3QXVXXnlvKEQhJncwVEFqbSIsImRhdGEiOnsidXNlciI6eyJpZCI6IjE0NCJ9fX0.Dn0q30ED459n7TK3L4zKIbfcB6_UKnutiWTp3N2MGfQ" -H "appid:dd4a7c" -H "appversion:2.0.1" -H "appcodeversion:1.2.8" -H "appplatform:ios" https://bb-app-platform.flywheelsites.com//wp-json/buddyboss/v1/activity/link-preview?url=https:%2F%2Fbb-app-platform.flywheelsites.com%2Fthe-new-day-wwe%2F
```
 Response
```
<iframe class="wp-embedded-content" sandbox="allow-scripts" security="restricted" style="position: absolute; clip: rect(1px, 1px, 1px, 1px);" title="&#8220;The New Day (WWE)&#8221; &#8212; BuddyBoss App Platform" src="https://bb-app-platform.flywheelsites.com/the-new-day-wwe/embed/#?secret=M1MeP1ELMD" data-secret="M1MeP1ELMD" width="200" height="200" frameborder="0" marginwidth="0" marginheight="0" scrolling="no"></iframe>
```
https://prntscr.com/1h16cff
https://www.loom.com/share/cf4d9a6efd1345b28fb211469cef66c4?sharedAppSource=personal_library

### How to test the changes in this Pull Request:

1. Go to BuddyBoss setting
2. Make site private and add some post url as public
3. Copy iframe url sent in preview api and paste in browser (incognito mode)
4. See error

### Proof Screenshots or Video



### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
